app/components/billing/payment.js

```diff
@@ -133,7 +133,8 @@ export default Component.extend({
             token: token.id,
             lastDigits: token.card.last4
           });
-          yield subscription.save();
+          const { clientSecret } = yield subscription.save();
+          this.stripe.handleStripePayment.perform(clientSecret);
```

Comment:
Same here, `yield` seems to be required here.

Reply:
When using `yield` here, subscription does not update after the payment is performed, while doing it this way does work.

Reply:
Would it make sense to investigate the root issue in this case? I'm afraid it could lead to some nasty bugs, because something seems to be wrong here.

Reply:
@artursmirnov could you please clarify 'something seems to be wrong' ?
cc @shairyar

Reply:
@sgolu6 `handleStripePayment` is an asynchronous task that handles Stripe payment (sends request to Stripe and waits for response). This method is _aimed_ to run asynchronously and be awaited to get result. In the current implementation, the code just sends the payment request to Stripe and continues to subscription update without waiting for results from Stripe. If it doesn't work as intended, then there might be some kind of a hidden error that causes it. And as long as this is quite sensible part (handling the payment), I though, it wouldn't hurt to be careful with it.

Reply:
As discussed internally with @shairyar, this is acceptable and we can leave it this way for now.

Reply:
Thank you @artursmirnov